src/thread.rs

```diff
@@ -501,6 +508,7 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
             if thread.state == ThreadState::Enabled {
                 if !self.yield_active_thread || id != self.active_thread {
                     self.active_thread = id;
+                    data_race.thread_set_active(self.active_thread);
```

Comment:
The AllocExtra functions for read, write and deallocate do not have access to the current active thread in its current location, and need the current id to perform the validation, and also the current thread name for reporting diagnostics. So either a local copy needs to be kept, or the associated thread metadata should be moved to a Rc<RefCell<..>> so that each AllocExtra for the data-race detector can access the value reference, (this will add the overhead for looking up a thread-id to a vector-idx for each access). Should I move to the Rc<RefCell<..>> model?

Reply:
Ah, so the problem is that we do not have access to the `MemoryExtra` here? Yeah, that's annoying...

Could you leave a comment explaining the situation? I have some ideas for rearranging these access hooks so that they *can* access `MemoryExtra` When/if that materializes, I'll ping you and maybe you can clean this up then. :)

Reply:
The currently active thread is stored in the ```Evaluator<'mir,'tcx>{ ..., threads: ThreadManager<..> .. }``` instead of ```MemoryExtra```, but the same issue occurs in either location, i'll add a comment now.

Reply:
Ah, fair -- but moving it to `MemoryExtra` would be easy (whether we put something into `Evaluator` or `MemoryExtra` is pretty arbitrary). So once `MemoryExtra` is available in these hooks, the problem can be solved cleanly.